sdk/include/platform/ibex/platform-dma.hh

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <compartment-macros.h>
+#include <cstdio>
+#include <utils.hh>
+
+namespace Ibex {
+    class PlatformDMA {
+        private:
+
+        struct DMAInterface
+        {
+            /**
+             * Below is DMA control register address: 
+             *  - includes start('b0), endianness conversion('b1 and 2) and reset bits('b3)
+             *  - bit of index 1 and 2 are for enabling 2 and 4 byte swaps respectively
+             *  - start bit is meant to be set at the end while programming
+             */
+            uint32_t control;
+            /**
+             * Below is DMA status register address 
+             *  - first bit refers to halted status
+             *  - 0 is for idle, and 1 for running
+             */
+            uint32_t status;
+            /**
+             * Below is DMA source address register address: 
+             *  - here source address is where FROM the DMA should transfer the data
+             *  - it can be either memory buffer or MMAP-ed I/O device (i.e. any peripheral or accelerator)
+             */
+            uint32_t sourceAddress;
+            /**
+             * Below is DMA target address register address: 
+             *  - here target address is where TO the DMA should transfer the data
+             *  - it can be either memory buffer or MMAP-ed I/O device (i.e. any peripheral or accelerator)
+             */
+            uint32_t targetAddress;
+            /**
+             * Below is the amount of data IN BYTES that DMA should transfer 
+             * between the source and a target 
+             */
+            uint32_t lengthInBytes;
+            /**
+             * Below is the amount of strides IN 4 BYTE WORDS for a source address. 
```

Comment:
- Strides probably want to be in bytes.  I suspect that interacts poorly with the desire to swap on 2- or 4-byte granularities.
- Zero stride probably should mean zero stride, or there should, at least, be a way of spelling zero-stride, for use with MMIO FIFOs.